### PR TITLE
[Merged by Bors] - feat: add `integral_eq_iff_of_ae_le`

### PR DESCRIPTION
**Pull request by RemyDegenne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Integral/Bochner.lean [hunk 1]
```diff
@@ -1271,6 +1271,20 @@ theorem integral_eq_zero_iff_of_nonneg_ae {f : α → ℝ} (hf : 0 ≤ᵐ[μ] f)
     simp only [Pi.zero_apply, ofReal_eq_zero]
   · exact (ENNReal.measurable_ofReal.comp_aemeasurable hfi.1.aemeasurable)
 
+lemma ae_eq_of_ae_le_of_integral_eq {f g : α → ℝ} (hf : Integrable f μ)
+    (hg : Integrable g μ) (h_le : f ≤ᵐ[μ] g) (h_eq : ∫ a, f a ∂μ = ∫ a, g a ∂μ) :
+    f =ᵐ[μ] g := by
```
Comment (fpvandoorn) [inline Mathlib/MeasureTheory/Integral/Bochner.lean:1276]: Can you make this an `iff` (with `h_eq`)?

**Comment by fpvandoorn** [comment]
I was thinking making the `iff` with the integrals being equal, so to have it the same shape as `MeasureTheory.integral_eq_zero_iff_of_nonneg_ae`. 
We could state it as `f =ᵐ[μ] g` iff the conjunction of these two statements, but I'm not sure if that will be super useful. I'll let you decide how to finally state this.

bors d+

**Comment by fpvandoorn** [comment]
Note: my second comment that some supporting lemmas could be used here to simplify the proof still stands

File: Mathlib/MeasureTheory/Integral/Bochner.lean (hunk 1 above)
Comment (RemyDegenne) [reply Mathlib/MeasureTheory/Integral/Bochner.lean]: Sorry, I did not see that second comment. I'll add those lemmas.

**Pull request closed without merging**